dali/pipeline/operators/crop/crop.h

```diff
@@ -1,174 +1,179 @@
-
-  // Output data type
-  DALIDataType output_type_;
-
-  // Output data layout
-  DALITensorLayout output_layout_;
-
-  USE_OPERATOR_MEMBERS();
-};
-
-}  // namespace dali
-
-#endif  // DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+#define DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+
+#include <utility>
+#include <vector>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+class CropAttr {
+ protected:
+  explicit inline CropAttr(const OpSpec &spec)
+      : image_type_(spec.GetArgument<DALIImageType>("image_type")),
+        C_(IsColor(image_type_) ? 3 : 1),
+        batch_size_{spec.GetArgument<int>("batch_size")} {
+    if (spec.name() != "Resize") {
+      vector<float> cropArgs = spec.GetRepeatedArgument<float>("crop");
+
+      DALI_ENFORCE(cropArgs[0] >= 0, "Crop height must be greater than zero. Received: " +
+                                         std::to_string(cropArgs[0]));
+      DALI_ENFORCE(cropArgs[1] >= 0, "Crop width must be greater than zero. Received: " +
+                                         std::to_string(cropArgs[1]));
+
+      crop_height_ = std::vector<int>(batch_size_, static_cast<int>(cropArgs[0]));
+      crop_width_ = std::vector<int>(batch_size_, static_cast<int>(cropArgs[1]));
+    }
+  }
+
+  std::pair<int, int> SetCropXY(const OpSpec &spec, const ArgumentWorkspace *ws,
+                                const Index dataIdx, int H, int W) {
+    DALI_ENFORCE(H >= crop_height_[dataIdx]);
+    DALI_ENFORCE(W >= crop_width_[dataIdx]);
+
+    auto crop_x_norm = spec.GetArgument<float>("crop_pos_x", ws, dataIdx);
+    auto crop_y_norm = spec.GetArgument<float>("crop_pos_y", ws, dataIdx);
+
+    DALI_ENFORCE(crop_y_norm >= 0.f && crop_y_norm <= 1.f,
+                 "Crop coordinates need to be in range [0.0, 1.0]");
+    DALI_ENFORCE(crop_x_norm >= 0.f && crop_x_norm <= 1.f,
+                 "Crop coordinates need to be in range [0.0, 1.0]");
+
+    const int crop_y = crop_y_norm * (H - crop_height_[dataIdx]);
+    const int crop_x = crop_x_norm * (W - crop_width_[dataIdx]);
+
+    return std::make_pair(crop_y, crop_x);
+  }
+
+  const vector<Index> CheckShapes(const SampleWorkspace *ws) {
```

Comment:
@szalpal This function was changed to be aligned with error messages across other operators, that explicitly used `shape[i]` for `i` in [0, 3). 

Also i reverted the 
```
 // enforce that all shapes match
    for (int i = 1; i < ws->NumInput(); ++i) {
    return input.shape();
      DALI_ENFORCE(input.SameShape(ws->Input<CPUBackend>(i)));
  }
}
```
loop that got somehow removed in previous change, but appeared during rebase and seemed important.